src/validation.cpp

```diff
@@ -2105,23 +2106,34 @@ bool static FlushStateToDisk(const CChainParams& chainparams, CValidationState &
     static int64_t nLastFlush = 0;
     std::set<int> setFilesToPrune;
     bool full_flush_completed = false;
+
+    // we delay pruning when one of the blockfilterindex is syncing
+    bool delay_pruning = false;
+    ForEachBlockFilterIndex([&delay_pruning](BlockFilterIndex& index) {
+        if (!index.IsSynced()) delay_pruning = true;
+    });
```

Comment:
Won't this effectively prevent all pruning until IBD completes? Seems important to prune as soon as all indexes are caught up to the given height...